mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala

```diff
@@ -99,19 +106,27 @@ object Keyword {
     Keyword("throw", isExpression = true),
     Keyword("implicit", isBlock = true, isTemplate = true),
     Keyword("return", isMethodBody = true),
+    Keyword("extends", leadingReverseTokens = Some(leadingTokensExtend)),
     Keyword("match"), // already implemented by CompletionPosition
     Keyword("case"), // already implemented by CompletionPosition and "case class"
     Keyword("override"), // already implemented by CompletionPosition
     Keyword("forSome"), // in-frequently used language feature
     Keyword("macro"), // in-frequently used language feature
     // The keywords below were left out in the first iteration of implementing keyword completions
     // since they appear in positions that are a bit more difficult to detect on the syntax tree.
-    Keyword("extends"),
     Keyword("with"),
     Keyword("catch"),
-    Keyword("extends"),
     Keyword("finally"),
     Keyword("then")
   )
 
+  private def leadingTokensExtend(leadingReverseTokens: Iterator[Token]): Boolean = {
+    leadingReverseTokens.filterNot(t => t.is[Token.Whitespace] || t.is[Token.EOF]).take(3).toList match {
+      // (class|trait|object) classname ext@@
+      case (_: Token.Ident) :: (_: Token.Ident) :: kw :: Nil =>
+        if (kw.is[Token.KwClass] || kw.is[Token.KwTrait] || kw.is[Token.KwObject]) true
```

Comment:
Could we also check if the last ident if it is a start of `extend`?

Reply:
Something along the lines of
```suggestion
        if (kw.is[Token.KwClass] || kw.is[Token.KwTrait] || kw.is[Token.KwObject] || kw.is[Token.RightParen]) "extends".startsWith(firstIdent.value)
```
`RightParen` case should help with the constructor args

Reply:
It will be tested at https://github.com/scalameta/metals/blob/c2c6b7077f46b9d30e9be3806691e792919bfa92/mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala#L49
so no need to test here :)